client/src/lib/components/DatasetForm/DatasetForm.svelte

```diff
@@ -150,6 +165,24 @@
     return typeof error === "string" && Boolean(error);
   };
 
+  // the tag error returned by yup could be a string, an array of string,  or an object with the same shape of Tag
+  const hasTagsError = (error: string | Tag[] | string[]): boolean => {
+    if (typeof error == "string" && error === "") return false;
+    if (typeof error == "string" && error !== "") {
+      return true;
+    }
+
+    const errorArray = error as unknown as Tag[];
+
+    if (
+      errorArray.findIndex((item) => item.id !== "" || item.name !== "") !== -1
+    ) {
+      return true;
+    }
+
+    return false;
+  };
```

Comment:
À mon avis on peut pas mal simplifier ça, car en réalité on ne veut afficher une erreur que quand on a une string non-vide. _Il me semble_ que les cas où on a des arrays ne sont pas traités / traitables, perso je ne sais pas bien à quoi ça correspond car le seul cas d'erreur peut provenir du `.min(1)` (non ?).

```suggestion
  // The tag error returned by yup could be a string, an array of string,  or an object with the same shape of Tag
  // But only the string can be non-empty (array min length constraint).
  $: hasTagsError = typeof $errors.tags === "string" && !!$errors.tags;
```